cmd/ipfs/main.go

```diff
@@ -473,6 +481,9 @@ func (i *cmdInvocation) setupInterruptHandler() {
 	sig := allInterruptSignals()
 
 	go func() {
+		// wait till the context is ready to be closed
+		<-ctx.ContextIsReadyToBeClosed
```

Comment:
So we have to either accept inability to kill the daemon during bootstrapping or use `os.Exit()`.